api/src/main/java/com/codeforcommunity/rest/ApiRouter.java

```diff
@@ -145,18 +197,19 @@ private void handleLogin(RoutingContext ctx) {
     if (isAuthorizedUser(request)) {
       response.putHeader("location", "/").setStatusCode(302).end();
     }
-
-    String username = "";
+    JsonObject body = ctx.getBodyAsJson();
```

Comment:
You can use Jackson to automatically marshal the body into a DTO object. This gives a standardized way of reading the body into two strings so that you dont have to have a try catch everytime you do this